addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxNotificationPlayer.java

```diff
@@ -0,0 +1,261 @@
+    final int volumeCommandTimeout = 4;
+
+    // We expect the media server to acknowledge a playlist command.
+    // An exception is thrown if the playlist command was not processed
+    // after the defined amount in [s]
+    final int playlistCommandTimeout = 5;
+
+    // Max length of the message in [s]. An exception is thrown if we did not
+    // receive a "stop" message from the media server.
+    final int notificationLengthTimeout = 90;
+
+    private Logger logger = LoggerFactory.getLogger(SqueezeBoxNotificationPlayer.class);
+    private SqueezeBoxPlayerState playerState;
+    private SqueezeBoxPlayerHandler squeezeBoxPlayerHandler;
+    private SqueezeBoxServerHandler squeezeBoxServerHandler;
+    private StringType uri;
+    private String mac;
+
+    boolean playlistModified;
+
+    private int notificationMessagePlaylistsIndex;
+
+    SqueezeBoxNotificationPlayer(SqueezeBoxPlayerHandler squeezeBoxPlayerHandler,
+            SqueezeBoxServerHandler squeezeBoxServerHandler, StringType uri) {
+        this.squeezeBoxPlayerHandler = squeezeBoxPlayerHandler;
+        this.squeezeBoxServerHandler = squeezeBoxServerHandler;
+        this.mac = squeezeBoxPlayerHandler.getMac();
+        this.uri = uri;
+        this.playerState = new SqueezeBoxPlayerState(squeezeBoxPlayerHandler);
+    }
+
+    void play() throws InterruptedException, SqueezeBoxTimeoutException {
+        if (squeezeBoxServerHandler == null) {
+            logger.warn("Server handler is null");
+            return;
+        }
+
+        setupPlayerForNotification();
+
+        try {
+            addNotificationMessageToPlaylist();
+            playNotification();
+        } finally {
+            if (playlistModified) {
+                // Mute the player to prevent any noise during the transition
+                // to previous state.
+                setVolume(0);
+                removeNotificationMessageFromPlaylist();
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        restorePlayerState();
+    }
+
+    private void setupPlayerForNotification() throws InterruptedException, SqueezeBoxTimeoutException {
+        logger.debug("Setting up player for notification");
+        if (!playerState.isPoweredOn()) {
+            logger.debug("Powering on the player");
+            squeezeBoxServerHandler.powerOn(mac);
+        }
+        if (playerState.isShuffling()) {
+            logger.debug("Turning off shuffle");
+            squeezeBoxServerHandler.setShuffleMode(mac, 0);
+        }
+        if (playerState.isRepeating()) {
+            logger.debug("Turning off repeat");
+            squeezeBoxServerHandler.setRepeatMode(mac, 0);
+        }
+        if (playerState.isPlaying()) {
+            squeezeBoxServerHandler.stop(mac);
+        }
+
+        int notificationVolume = squeezeBoxPlayerHandler.getNotificationSoundVolume().intValue();
+        setVolume(notificationVolume);
```

Comment:
I wonder if it would make sense to log INFO or DEBUG if notificationVolume is 0.  It might help with some head-scratching when the notification message is not heard.

Reply:
I´ll add a WARN message, as this will defenitely cause-scratching and forum entries.